src/contracts.nit

```diff
@@ -453,50 +614,135 @@ class MEnsure
 		return n_signature.adapt_to_ensurecondition
 	end
 
-	redef fun adapt_block_to_contract(v: ContractsVisitor, n_mpropdef: AMethPropdef)
+	redef fun adapt_method_to_contract(v: ContractsVisitor, n_mpropdef: AMethPropdef, called_method: MMethod)
 	do
 		var callsite = v.ast_builder.create_callsite(v.toolcontext.modelbuilder, n_mpropdef, self, true)
 		var n_self = new ASelfExpr
 		# argument to call the contract method
 		var args = n_mpropdef.n_signature.make_parameter_read(v.ast_builder)
-		# Inject the variable result
-		# The cast is always safe because the online adapted method is the contract facet
-
 		var actual_block = n_mpropdef.n_block
 		# never happen. If it's the case the problem is in the contract facet building
 		assert actual_block isa ABlockExpr
-
 		var ret_type = n_mpropdef.mpropdef.mproperty.intro.msignature.return_mtype
 		if ret_type != null then
+			# Inject the variable result
 			var result_var = inject_result(v, n_mpropdef, ret_type)
 			# Expr to read the result variable
 			var read_result = v.ast_builder.make_var_read(result_var, ret_type)
 			var return_expr = actual_block.n_expr.pop
 			# Adding the read return to argument
 			args.add(read_result)
 			var n_callcontract = v.ast_builder.make_call(n_self,callsite,args)
-			actual_block.add_all([n_callcontract,return_expr])
+			actual_block.add_all([v.encapsulated_contract_call(n_mpropdef, [n_callcontract]), return_expr])
 		else
 			# build the call to the contract method
 			var n_callcontract = v.ast_builder.make_call(n_self,callsite,args)
-			actual_block.add n_callcontract
+			actual_block.add v.encapsulated_contract_call(n_mpropdef, [n_callcontract])
 		end
 	end
 end
 
+# A invariant contract representation
+# This method check if the Object is in a valid state.
```

Comment:
```suggestion
# This method checks if the Object is in a valid state.
```

But also, the comment is on a class, not a method? Is it misplaced?

Reply:
I wanted to indicate the objective of the invariants but it is not really useful here.